### DISCUSSION TITLE
I have a a couple of functions that make a lot of calls to the RPC but I don't feel they are getting batch

### DISCUSSION THREAD

**Discussion by guillermoriv** [wevm/wagmi#2752] [Q&A] [answered]
I have a couple of functions that fetch data from the RPC's based on the network we are on the app, I was doing 324 requests total with `JsonBatchProvider` from ethers and after migrating to viem/wagmi, I am doing 1024 requests total (after fetching everything), does the batch works differently in wagmi?

```typescript
export const { chains, publicClient } = configureChains(
  Object.values(networks),
  [
    publicProvider(),
    jsonRpcProvider({
      rpc: chain => ({ http: chain.rpcUrls.default.http[0] }),
    }),
  ],
  {
    pollingInterval: POLLING_INTERVAL,
    batch: { multicall: true },
  },
);
```

This is my code, I don't know if it has to do with the `publicProvider`, still I have a fallback setup with the `jsonRcpProvider`.

This is my current total requests:

![image](https://github.com/wagmi-dev/wagmi/assets/61250854/d5247d1a-42c7-44be-bae7-2e630ceaed22)

And I only can see one `eth_call` and not a batch:

![image](https://github.com/wagmi-dev/wagmi/assets/61250854/6a2d12eb-ffef-43a9-bc23-17406cb8f08b)

**Comment by jxom** [comment, marked as answer]
wagmi utilises viems [Multicall Aggregation](https://viem.sh/docs/clients/public.html#eth-call-aggregation-via-multicall) to batch `eth_call` requests, which is different to ethers’ Batch JSON-RPC approach. While viem does [support Batch JSON-RPC](https://viem.sh/docs/clients/transports/http.html#batch-json-rpc), this is not enabled by default in wagmi because browsers are generally pretty good at invoking several thousand concurrent requests with HTTP/2, plus RPC Providers commonly only support up to 1000 requests at a time anyway.

The Multicall Aggregation batch config is currently set up to batch in 32ms intervals, but you can [configure this](https://wagmi.sh/react/providers/configuring-chains#batchmulticall-optional) if you want to something higher at the cost of response latency. Ethers defaults to 250ms for Batch JSON-RPC, so I suspect that’s why it’s lower. I wouldn’t recommend setting it that high though.

**Reply by guillermoriv** [reply]
Thanks @jxom
